src/generate_transactions.rs

```diff
@@ -41,18 +54,27 @@ pub async fn send_continuous_transactions(
     Ok(())
 }
 
-async fn generate_and_send_transfer(tx_manager: &TransactionManager) -> Result<()> {
-    // Generate a new wallet for the recipient
-    let recipient_wallet = Wallet::new(&mut rand::thread_rng());
-    let to = recipient_wallet.address();
+async fn generate_and_send_transfer(
+    provider: Arc<Provider<Http>>,
+    mut tx_manager: TransactionManager,
+    level: u64,
+) -> Result<()> {
+    for i in (1..=level).rev() {
+        // Generate a new wallet for the recipient
+        let recipient_wallet = Wallet::new(&mut rand::thread_rng());
+        let to = recipient_wallet.address();
 
-    // Define the transfer
-    let tx = TransactionRequest::new()
-        .to(to)
-        .value(1e6 as u64)
-        .from(tx_manager.get_address());
+        // Transfer `total - 1e12` to the recipient and leave 1e12 as tx fee
```

Comment:
This math doesn't leave enough for follow on transactions. Maybe paramaterize amount, then have it be some amount just shy of the sending wallets current balance.